core/src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerRule.java

```diff
@@ -0,0 +1,264 @@
+     */
+    static CircuitBreakerRule onStatusClass(Iterable<HttpStatusClass> statusClasses) {
+        return builder().onStatusClass(statusClasses).thenFailure();
+    }
+
+    /**
+     * Returns newly created {@link CircuitBreakerRule} that will report a {@link Response} as a failure,
+     * if the class of the response status is {@link HttpStatusClass#SERVER_ERROR}.
+     */
+    static CircuitBreakerRule onServerErrorStatus() {
+        return builder().onServerErrorStatus().thenFailure();
+    }
+
+    /**
+     * Returns newly created {@link CircuitBreakerRule} that will report a {@link Response} as a failure,
+     * if the response status is one of the specified {@link HttpStatus}es.
+     */
+    static CircuitBreakerRule onStatus(HttpStatus... statuses) {
+        return builder().onStatus(statuses).thenFailure();
+    }
+
+    /**
+     * Returns newly created {@link CircuitBreakerRule} that will report a {@link Response} as a failure,
+     * if the response status is one of the specified {@link HttpStatus}es.
+     */
+    static CircuitBreakerRule onStatus(Iterable<HttpStatus> statuses) {
+        return builder().onStatus(statuses).thenFailure();
+    }
+
+    /**
+     * Returns newly created {@link CircuitBreakerRule} that will report a {@link Response} as a failure,
+     * if the specified {@code statusFilter} returns {@code true}.
+     */
+    static CircuitBreakerRule onStatus(Predicate<? super HttpStatus> statusFilter) {
+        return builder().onStatus(statusFilter).thenFailure();
+    }
+
+    /**
+     * Returns newly created {@link CircuitBreakerRule} that will report a {@link Response} as a failure,
+     * if an {@link Exception} is raised and that is instance of the specified {@code exception}.
+     */
+    static CircuitBreakerRule onException(Class<? extends Throwable> exception) {
+        return builder().onException(exception).thenFailure();
+    }
+
+    /**
+     * Returns newly created {@link CircuitBreakerRule} that will report a {@link Response} as a failure,
+     * if an {@link Exception} is raised and the specified {@code exceptionFilter} returns {@code true}.
+     */
+    static CircuitBreakerRule onException(Predicate<? super Throwable> exceptionFilter) {
+        return builder().onException(exceptionFilter).thenFailure();
+    }
+
+    /**
+     * Returns newly created {@link CircuitBreakerRule} that will report a {@link Response} as a failure,
+     * if an {@link Exception} is raised.
+     */
+    static CircuitBreakerRule onException() {
+        return builder().onException().thenFailure();
+    }
+
+    /**
+     * Returns newly created {@link CircuitBreakerRule} that will report a {@link Response} as a failure,
+     * if an {@link UnprocessedRequestException} which means that the request has not been processed by
+     * the server is raised.
+     */
+    static CircuitBreakerRule onUnprocessed() {
+        return builder().onUnprocessed().thenFailure();
```

Comment:
I think the default should be `thenIgnore` for `onUnprocessed`.

Reply:
Ignoring on unprocessed seems default behavior.
However, other factory methods build a rule with `thenFailure()`.
Probably better to remove this factory to reduce confusion?

Reply:
+1 for removing. 😄